gems/sorbet/lib/constant_cache.rb

```diff
@@ -138,7 +138,7 @@ def name_by_class(klass)
         rescue LoadError => err
           puts "Got #{err.class} when trying to get nested name #{name}::#{nested}"
           next
-        rescue NameError, ArgumentError => err
+        rescue NameError, ArgumentError, RuntimeError => err
```

Comment:
`RuntimeError` is very general, can we be more specific here?
What error did the gem that you had tried have?

Reply:
The actual error class was gem specific and the lowest plain old Ruby ancestor appears to be `RuntimeError`.

![Screen Shot 2019-09-20 at 7 27 18 PM](https://user-images.githubusercontent.com/2396007/65363973-cb344800-dbdc-11e9-9cbb-149a5e407b1b.png)

Reply:
I agree it's pretty bad to be catching such a general error type here.

But I think it's also quite bad to have Sorbet usage become totally broken by some nonsense happening in a Gem you use. For me personally, having run into similar issues to this 3 or 4 times over the last 2 months, this has been by far my biggest pain point with a Sorbet; otherwise I've found using it pretty great.

Reply:
@mgriisser the approach we've been using for this so far is to specify either via annotations in the `Gemfile` that certain things shouldn't be required (`require: false`) or with custom logic in `gems/sorbet/lib/gem_loader.rb`

Is it feasible to fix the problems you were seeing by using one of those two techniques?

Reply:
I tried both of those but wasn't able to address my current issue with either, though I'm not sure I fully understand what `gem_loader` is doing. Would this be because the error is occurring during `hidden-definitions` and now the `gems` phase?